### PR TITLE
[ros422] Look for OpLookupReg test case

### PR DESCRIPTION
**Pull request by AbnerZheng**
## Overview
Look for OpLookupReg test case

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/ROS-422

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

**Comment by AbnerZheng** [comment]
@tymm PTAL

File: roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala [hunk 1]
```diff
@@ -619,4 +619,59 @@ class VmSpec extends FlatSpec with Matchers {
...
+      * (free [〈id〉 …] 〈body〉) ⇒ result of 〈body〉
+      * The free form informs the compiler that the 〈id〉 are to be treated as free in 〈body〉;
+      * otherwise, the compiler will issue warnings that the 〈id〉 do not have a compile-time binding.
+      * And then it will be found in somewhere.
+      *
+      */
+    val codevec = Seq(
+      OpAlloc(n = 1),
+      OpImmediateLitToArg(literal = `1`, arg = 0),
+      OpLookupToReg(lit = `1`, reg = trgt),
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Please do `lit = 1` here. We use the backtick variables only for `literal`.

File: roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala [hunk 2]
```diff
@@ -619,4 +619,59 @@ class VmSpec extends FlatSpec with Matchers {
...
+
+    val code = Code(litvec = Seq(Niv, RblString("cat")), codevec = codevec)
+
+    val ctxt        = Ctxt(code, rtnCtxt, LocRslt)
+    // todo we set RequestExpr's parent and meta to the most outer case for simplicity, actually there would be RequestExprSBO and RequestExprMeta
+    val requestExpr = new RequestExpr(Seq.empty[Ob])
+
+    // we need RBLtopenv to end the recurse `lookup`
+    requestExpr.parent = new RBLtopenv
+    requestExpr.meta = topMeta
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: I just checked - in this example the `meta` of the `RequestExpr` is not `TopMeta`.
Just do `requestExpr.meta = Meta.empty` in order to not confuse our future selves.

Comment (AbnerZheng) [reply roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Done

File: roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala (hunk 1 above)
Comment (AbnerZheng) [reply roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Done

File: roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala [hunk 3]
```diff
@@ -619,4 +619,59 @@ class VmSpec extends FlatSpec with Matchers {
...
+    val codevec = Seq(
+      OpAlloc(n = 1),
+      OpImmediateLitToArg(literal = `1`, arg = 0),
+      OpLookupToReg(lit = `1`, reg = trgt),
+      OpXmit(unwind = false, next = true, nargs = 1)
+    )
+
+    val rtnCtxt = Ctxt.outstanding(1)
+
+    val code = Code(litvec = Seq(Niv, RblString("cat")), codevec = codevec)
```
[the remark this answers is not in the document]
Comment (AbnerZheng) [reply roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Done

**Comment by AbnerZheng** [comment]
@tymm PTAL

File: roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala [hunk 4]
```diff
@@ -3,16 +3,16 @@ package coop.rchain.roscala
...
 import coop.rchain.roscala.prim.rblfloat.flPlus
 import org.scalatest.{FlatSpec, Matchers}
 
-import scala.collection.mutable
-
 class VmSpec extends FlatSpec with Matchers {
 
   val globalEnv = new GlobalEnv()
+  // this is defined in BigBang.cc#InitBuiltinObs
+  val topMeta = Meta.empty
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Please remove since we don't need that for now.

Comment (AbnerZheng) [reply roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala]: Done.

**Pull request merged**